packages/client-assertion-validation/test/validation.test.ts

```diff
@@ -203,148 +202,127 @@ describe("validation test", () => {
       expect(errors).toBeDefined();
       expect(errors).toHaveLength(1);
       expect(errors![0]).toEqual(invalidAudience());
     });
 
-    it("invalidAudience - wrong entry as 1-item array", () => {
-      const a = getMockClientAssertion({
+    it("invalidAudience - wrong entry as 1-item array", async () => {
+      const { jws } = await getMockClientAssertion({
         customHeader: {},
         standardClaimsOverride: { aud: ["random"] },
         customClaims: {},
       });
-      const { errors } = verifyClientAssertion(a, undefined);
+      const { errors } = verifyClientAssertion(jws, undefined);
       expect(errors).toBeDefined();
       expect(errors).toHaveLength(1);
       expect(errors![0]).toEqual(invalidAudience());
     });
 
-    it("invalidAudienceFormat - comma-separated strings", () => {
-      const a = getMockClientAssertion({
+    it("invalidAudienceFormat - comma-separated strings", async () => {
+      const { jws } = await getMockClientAssertion({
         customHeader: {},
         standardClaimsOverride: { aud: "test.interop.pagopa.it, other-aud" },
         customClaims: {},
       });
-      const { errors } = verifyClientAssertion(a, undefined);
+      const { errors } = verifyClientAssertion(jws, undefined);
       expect(errors).toBeDefined();
       expect(errors).toHaveLength(1);
       expect(errors![0]).toEqual(invalidAudienceFormat());
     });
 
-    it("invalidAudience - wrong entries", () => {
-      const a = getMockClientAssertion({
+    it("invalidAudience - wrong entries", async () => {
+      const { jws } = await getMockClientAssertion({
         customHeader: {},
         standardClaimsOverride: { aud: ["wrong-audience1, wrong-audience2"] },
         customClaims: {},
       });
-      const { errors } = verifyClientAssertion(a, undefined);
+      const { errors } = verifyClientAssertion(jws, undefined);
       expect(errors).toBeDefined();
       expect(errors).toHaveLength(1);
       expect(errors![0]).toEqual(invalidAudience());
     });
 
-    it("invalidAudience - missing entry", () => {
-      const a = getMockClientAssertion({
+    it("invalidAudience - missing entry", async () => {
+      const { jws } = await getMockClientAssertion({
         customHeader: {},
         standardClaimsOverride: {
           aud: ["test.interop.pagopa.it"],
         },
         customClaims: {},
       });
-      const { errors } = verifyClientAssertion(a, undefined);
-      expect(errors).toBeDefined();
-      expect(errors).toHaveLength(1);
-      expect(errors![0]).toEqual(invalidAudience());
-    });
-
-    it("unexpectedClientAssertionPayload", () => {
-      const key = crypto.generateKeyPairSync("rsa", {
-        modulusLength: 2048,
-      }).privateKey;
-
-      const options: jwt.SignOptions = {
-        header: {
-          kid: generateId(),
-          alg: "RS256",
-        },
-      };
-      const jws = jwt.sign("actualPayload", key, options);
-
```

Comment:
Is there a way to keep such test?

Reply:
I re-added jsonwebtoken as devDep to re-add this test, since with jose it would fail while signing the test assertion